salt/utils/gitfs.py

```diff
@@ -2829,17 +2834,24 @@ def file_hash(self, load, fnd):
                                         load['saltenv'],
                                         '{0}.hash.{1}'.format(relpath,
                                                               self.opts['hash_type']))
-        if not os.path.isfile(hashdest):
-            if not os.path.exists(os.path.dirname(hashdest)):
-                os.makedirs(os.path.dirname(hashdest))
-            ret['hsum'] = salt.utils.hashutils.get_hash(path, self.opts['hash_type'])
-            with salt.utils.files.fopen(hashdest, 'w+') as fp_:
-                fp_.write(ret['hsum'])
-            return ret
-        else:
+        try:
             with salt.utils.files.fopen(hashdest, 'rb') as fp_:
                 ret['hsum'] = fp_.read()
             return ret
+        except IOError as exc:
+            if exc.errno != errno.ENOENT:
+                raise exc
+
+        try:
+            os.makedirs(os.path.dirname(hashdest))
+        except OSError as exc:
+            if exc.errno != errno.EEXIST:
+                raise exc
+
+        ret['hsum'] = salt.utils.get_hash(path, self.opts['hash_type'])
```

Comment:
This should be `salt.utils.hashutils.get_hash`

Reply:
Ah, I forgot to run the check for the moved utils. Thanks!